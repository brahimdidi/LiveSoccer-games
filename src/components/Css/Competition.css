:root {
  --blue: #60707f;
  --white: #ffffff;
  --skyBlue: #61dafb;
  --grey: rgb(217, 206, 206);
}
.text-center {
  text-align: center;
}

.Competitions-container {
  width: 80%;
  max-width: 600px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  gap: 20px;
  align-self: center;
}
.display-none {
  display: none;
};
.flex {
display: flex;
}
.competition-container {
  display: flex;
  flex-direction: column;
  background: rgb(39,39,39,1);
  border: 1px solid black;
  border-radius: 5px;
  padding: 5px;
  color: white;
}
.competition-show-list,.competition-list {
  margin-top: 10px;
}
.competition-list li {
  transition: 1s;
}