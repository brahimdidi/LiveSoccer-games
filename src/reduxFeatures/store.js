import { configureStore } from '@reduxjs/toolkit';
import allGamesReducer from './allGamesSlice';

const store = configureStore({
  reducer: {
    allGames: allGamesReducer,
  },
});

export default store;
