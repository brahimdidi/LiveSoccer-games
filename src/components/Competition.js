/*eslint-disable */
import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { IoIosArrowDropdown } from 'react-icons/io';
import { selectAllGames, fetchAllGames } from '../reduxFeatures/allGamesSlice';
import './Css/Competition.css';

function Competition() {
  // select all games from the main state.
  const allGames = useSelector(selectAllGames);
  const dispatch = useDispatch();
  // check if the state is empty , fetch data only once
  useEffect(() => {
    if (!allGames.length)dispatch(fetchAllGames());
  }, [dispatch]);
  // set the name fo the league clicked through useLocation object
  const location = useLocation();
  if (location.state) {
    var { league } = location.state;
    var showCompetition = allGames.filter((e) => e.competition === league);
  }
  // here we filter the most popular competitions to show.
  const friendlies = allGames.filter((e) => e.competition === 'INTERNATIONAL: Club Friendlies');
  const nationsLeague = allGames.filter((e) => e.competition === 'UEFA NATIONS LEAGUE A: Group Stage');
  const premierLeague = allGames.filter((e) => e.competition === 'ENGLAND: Premier League');
  const laLiga = allGames.filter((e) => e.competition === 'SPAIN: La Liga');
  const serieA = allGames.filter((e) => e.competition === 'ITALY: Serie A');
  const afcCup = allGames.filter((e) => e.competition === 'AFC CUP: Group Stage');
  // this function returns the jsx needed to render the UI
  const itemJsx = (league) => (
    <li key={league.id}>
      <h2 className="date text-center">{league.date}</h2>
      <div className="title-time">
        <h3>{league.title}</h3>
        <span>{league.time}</span>
      </div>

    </li>
  );

  const leagueEmpty = (e) => (
    <li key={34} className="league-empty">
      There is no
      {' '}
      {e}
      {' '}
      games
      yet!
    </li>
  );
  // useState to toggle the show more button
  const [showNationsLeague, setNationsLeague] = useState(false);
  const [showFriendlies, setFriendlies] = useState(false);
  const [showLaLiga, setLaLiga] = useState(false);
  const [showSerieA, setSerieA] = useState(false);
  const [showPremierLeague, setPremierLeague] = useState(false);
  const [showAfcCup, setAfcCup] = useState(false);
  //  render the UI : if the location object is null  render the leagues filter
  // form the state, else we only render the league which the user chose to see
  return (
    <>
      {!location.state
     && (
     <section className="Competitions-container">
       <div className="competition-container">
         <div className="league-name">
           <h1>Nations league (lastest games)</h1>
           <IoIosArrowDropdown onClick={() => setNationsLeague(!showNationsLeague)} />
         </div>
         <ul className={`competition-list ${showNationsLeague ? 'flex' : 'display-none'}`}>
           {nationsLeague.length ? ((nationsLeague.map((league) => (itemJsx(league))))) : leagueEmpty('Premier League') }
         </ul>
       </div>
       <div className="competition-container">
         <div className="league-name">
           <h1 className="league-name">La liga (lastest games)</h1>
           <IoIosArrowDropdown onClick={() => setLaLiga(!showLaLiga)} />
         </div>
         <ul className={`competition-list ${showLaLiga ? 'flex' : 'display-none'}`}>
           {laLiga.length ? ((laLiga.map((league) => (itemJsx(league))))) : leagueEmpty('La liga') }
         </ul>
       </div>
       <div className="competition-container">
         <div className="league-name">
           <h1 className="league-name">Serie A (lastest games)</h1>
           <IoIosArrowDropdown onClick={() => setSerieA(!showSerieA)} />
         </div>

         <ul className={`competition-list ${showSerieA ? 'flex' : 'display-none'}`}>

           {serieA.length ? ((serieA.map((league) => (itemJsx(league))))) : leagueEmpty('Serie A') }
         </ul>
       </div>
       <div className="competition-container">
         <div className="league-name">
           <h1>Premier League (lastest games)</h1>
           <IoIosArrowDropdown onClick={() => setPremierLeague(!showPremierLeague)} />
         </div>
         <ul className={`competition-list ${showPremierLeague ? 'flex' : 'display-none'}`}>

           {premierLeague.length ? ((premierLeague.map((league) => (itemJsx(league))))) : leagueEmpty('Nations League') }
         </ul>
       </div>
       <div className="competition-container">
         <div className="league-name">
           <h1>Club Friendlies (lastest games)</h1>
           <IoIosArrowDropdown onClick={() => setFriendlies(!showFriendlies)} />
         </div>
         <ul className={`competition-list ${showFriendlies ? 'flex' : 'display-none'}`}>

           {friendlies.length ? ((friendlies.map((league) => (itemJsx(league))))) : leagueEmpty('Friendlies ') }
         </ul>
       </div>
       <div className="competition-container">
         <div className="league-name">
           <h1>AFC CUP (lastest games)</h1>
           <IoIosArrowDropdown onClick={() => setAfcCup(!showAfcCup)} />
         </div>
         <ul className={`competition-list ${showAfcCup ? 'flex' : 'display-none'}`}>

           {afcCup.length ? ((afcCup.map((league) => (itemJsx(league))))) : leagueEmpty('AFC CUP ') }
         </ul>
       </div>
     </section>
     )}
      {location.state && (
      <section className="Competitions-container">
        <div className="competition-container competition-show">
          <div className="league-name">
            <h1>
              {league}
              {' '}
              (lastest games)
            </h1>
          </div>
          <ul className="competition-show-list flex">
            {showCompetition.length
              ? ((showCompetition.map((league) => (itemJsx(league)))))
              : leagueEmpty(showCompetition.competition) }
          </ul>
        </div>
      </section>
      )}
      {!allGames.length && <h1 className="text-danger">Check your internet Again</h1>}

    </>
  );
}

export default Competition;
