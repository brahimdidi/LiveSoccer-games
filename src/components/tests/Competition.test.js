import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../reduxFeatures/store';
import Competition from '../Competition';

describe('Competition Test', () => {
  it('renders correctly', () => {
    const CompetitionTest = renderer.create(
      <Provider store={store}>
        <Router>
          <Competition />
        </Router>
      </Provider>,
    ).toJSON();
    expect(CompetitionTest).toMatchSnapshot();
  });
});
