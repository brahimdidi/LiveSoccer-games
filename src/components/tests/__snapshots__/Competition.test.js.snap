// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllGames Test renders correctly 1`] = `
<div
  className="allGames-app"
>
  <nav
    className="allGames-nav"
  >
    <button
      className="btn btn-active"
      onClick={[Function]}
      type="button"
    >
      All Games
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      TodayGames
    </button>
  </nav>
  <ul
    className="games-container"
  >
    <h1
      className="text-danger"
    >
      Check your internet Again
    </h1>
  </ul>
</div>
`;

exports[`Competition Test renders correctly 1`] = `
Array [
  <section
    className="Competitions-container"
  >
    <div
      className="competition-container"
    >
      <div
        className="league-name"
      >
        <h1>
          Nations league (lastest games)
        </h1>
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M147.6 210.7c-7.5 7.5-7.5 19.8 0 27.3l95.7 95.4c7.3 7.3 19.1 7.5 26.6.6l94.3-94c3.8-3.8 5.7-8.7 5.7-13.7 0-4.9-1.9-9.9-5.6-13.6-7.5-7.5-19.7-7.6-27.3 0l-81 79.8-81.1-81.9c-7.5-7.5-19.7-7.5-27.3.1z"
          />
          <path
            d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm332.4-124.4C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256s18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6z"
          />
        </svg>
      </div>
      <ul
        className="competition-list display-none"
      >
        <li
          className="league-empty"
        >
          There is no
           
          Premier League
           
          games yet!
        </li>
      </ul>
    </div>
    <div
      className="competition-container"
    >
      <div
        className="league-name"
      >
        <h1
          className="league-name"
        >
          La liga (lastest games)
        </h1>
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M147.6 210.7c-7.5 7.5-7.5 19.8 0 27.3l95.7 95.4c7.3 7.3 19.1 7.5 26.6.6l94.3-94c3.8-3.8 5.7-8.7 5.7-13.7 0-4.9-1.9-9.9-5.6-13.6-7.5-7.5-19.7-7.6-27.3 0l-81 79.8-81.1-81.9c-7.5-7.5-19.7-7.5-27.3.1z"
          />
          <path
            d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm332.4-124.4C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256s18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6z"
          />
        </svg>
      </div>
      <ul
        className="competition-list display-none"
      >
        <li
          className="league-empty"
        >
          There is no
           
          La liga
           
          games yet!
        </li>
      </ul>
    </div>
    <div
      className="competition-container"
    >
      <div
        className="league-name"
      >
        <h1
          className="league-name"
        >
          Serie A (lastest games)
        </h1>
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M147.6 210.7c-7.5 7.5-7.5 19.8 0 27.3l95.7 95.4c7.3 7.3 19.1 7.5 26.6.6l94.3-94c3.8-3.8 5.7-8.7 5.7-13.7 0-4.9-1.9-9.9-5.6-13.6-7.5-7.5-19.7-7.6-27.3 0l-81 79.8-81.1-81.9c-7.5-7.5-19.7-7.5-27.3.1z"
          />
          <path
            d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm332.4-124.4C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256s18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6z"
          />
        </svg>
      </div>
      <ul
        className="competition-list display-none"
      >
        <li
          className="league-empty"
        >
          There is no
           
          Serie A
           
          games yet!
        </li>
      </ul>
    </div>
    <div
      className="competition-container"
    >
      <div
        className="league-name"
      >
        <h1>
          Premier League (lastest games)
        </h1>
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M147.6 210.7c-7.5 7.5-7.5 19.8 0 27.3l95.7 95.4c7.3 7.3 19.1 7.5 26.6.6l94.3-94c3.8-3.8 5.7-8.7 5.7-13.7 0-4.9-1.9-9.9-5.6-13.6-7.5-7.5-19.7-7.6-27.3 0l-81 79.8-81.1-81.9c-7.5-7.5-19.7-7.5-27.3.1z"
          />
          <path
            d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm332.4-124.4C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256s18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6z"
          />
        </svg>
      </div>
      <ul
        className="competition-list display-none"
      >
        <li
          className="league-empty"
        >
          There is no
           
          Nations League
           
          games yet!
        </li>
      </ul>
    </div>
    <div
      className="competition-container"
    >
      <div
        className="league-name"
      >
        <h1>
          Club Friendlies (lastest games)
        </h1>
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M147.6 210.7c-7.5 7.5-7.5 19.8 0 27.3l95.7 95.4c7.3 7.3 19.1 7.5 26.6.6l94.3-94c3.8-3.8 5.7-8.7 5.7-13.7 0-4.9-1.9-9.9-5.6-13.6-7.5-7.5-19.7-7.6-27.3 0l-81 79.8-81.1-81.9c-7.5-7.5-19.7-7.5-27.3.1z"
          />
          <path
            d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm332.4-124.4C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256s18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6z"
          />
        </svg>
      </div>
      <ul
        className="competition-list display-none"
      >
        <li
          className="league-empty"
        >
          There is no
           
          Friendlies 
           
          games yet!
        </li>
      </ul>
    </div>
    <div
      className="competition-container"
    >
      <div
        className="league-name"
      >
        <h1>
          AFC CUP (lastest games)
        </h1>
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M147.6 210.7c-7.5 7.5-7.5 19.8 0 27.3l95.7 95.4c7.3 7.3 19.1 7.5 26.6.6l94.3-94c3.8-3.8 5.7-8.7 5.7-13.7 0-4.9-1.9-9.9-5.6-13.6-7.5-7.5-19.7-7.6-27.3 0l-81 79.8-81.1-81.9c-7.5-7.5-19.7-7.5-27.3.1z"
          />
          <path
            d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm332.4-124.4C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256s18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6z"
          />
        </svg>
      </div>
      <ul
        className="competition-list display-none"
      >
        <li
          className="league-empty"
        >
          There is no
           
          AFC CUP 
           
          games yet!
        </li>
      </ul>
    </div>
  </section>,
  <h1
    className="text-danger"
  >
    Check your internet Again
  </h1>,
]
`;
